servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsClient.java

```diff
@@ -974,8 +972,9 @@ private static boolean shouldRevokeState(final Throwable t) {
         // ISE => Subscriber exceptions (downstream of retry)
         return t instanceof SrvAddressRemovedException || t instanceof IllegalStateException ||
                 t instanceof ClosedDnsServiceDiscovererException || (NX_DOMAIN_INVALIDATES &&
-                t.getCause() instanceof DnsErrorCauseException &&
-                NXDOMAIN.equals(((DnsErrorCauseException) t.getCause()).getCode()));
+                // string matching is done on purpose to avoid the hard Netty dependency
+                (t.getCause() != null && t.getCause().getClass().getSimpleName().contains("DnsErrorCauseException")) &&
```

Comment:
Consider using the full name match (not simple) with `equals` instead of `contains`